lib/neo4j/active_node/query/query_proxy_methods.rb

```diff
@@ -39,6 +39,7 @@ def order_property
 
         # @return [Integer] number of nodes of this class
         def count(distinct = nil, target = nil)
+          return 0 if unpersisted_start_object?
```

Comment:
Returning `0` feels like we've made a query and gotten an answer.  Maybe this should be `nil`?